### PR TITLE
Fix/user secrets

### PR DESCRIPTION
**Pull request by nzikalala**

### CODE REVIEW

File: HRIS.Models/AuthManagement.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
```
Comment (EduanO) [inline HRIS.Models/AuthManagement.cs:6]:
```suggestion
```

File: HRIS.Models/SMTPSettings.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
```
Comment (EduanO) [inline HRIS.Models/SMTPSettings.cs:6]:
```suggestion
```

File: HRIS.Services.Tests/Helpers/EmailHelperUnitTests.cs [hunk 3]
```diff
@@ -5,18 +5,30 @@
...
 using System.Linq.Expressions;
+using HRIS.Models;
+using Microsoft.Extensions.Options;
 
 namespace HRIS.Services.Tests.Helpers;
 
 public class EmailHelperUnitTests
 {
     private readonly Mock<IUnitOfWork> _db;
+    /*private readonly Mock<IOptions<SMTPSettings>> _smtp;*/
```
Comment (EduanO) [inline HRIS.Services.Tests/Helpers/EmailHelperUnitTests.cs:16]:
```suggestion
```

File: HRIS.Services.Tests/Helpers/EmailHelperUnitTests.cs [hunk 4]
```diff
@@ -5,18 +5,30 @@
...
+        var _smtp = Options.Create(new SMTPSettings{
+            Name = "test",
+            Host = "testHost",
+            Mail = "testMail@retrorabbit.co.za",
+            Password = "password",
+            Port = "54"
+        });
+
+        _helper = new EmailHelper(_db.Object, _smtp);
+
```
Comment (EduanO) [inline HRIS.Services.Tests/Helpers/EmailHelperUnitTests.cs:31]:
```suggestion
        var smtp = Options.Create(new SMTPSettings{
            Name = "test",
            Host = "testHost",
            Mail = "testMail@retrorabbit.co.za",
            Password = "password",
            Port = "54"
        });

        _helper = new EmailHelper(_db.Object, smtp);
```

File: RR.App/Program.cs [hunk 5]
```diff
@@ -16,13 +16,16 @@ namespace RR.App
...
 
         public static async Task Main(string[] args)
         {
             var builder = WebApplication.CreateBuilder(args);
             var configuration = builder.Configuration;
 
+            builder.Services.Configure<AuthManagement>(builder.Configuration.GetSection("AuthManagement"));
+            builder.Services.Configure<SMTPSettings>(builder.Configuration.GetSection("SMTP"));
+
+
```
Comment (EduanO) [inline RR.App/Program.cs:28]:
```suggestion

```

**Review by EduanO — CHANGES_REQUESTED** [review]
Minor code clean-up but overall happy with implementation

**Review by MathildaShongwe15 — DISMISSED** [review]

**Review by ReitumetseMoloto — DISMISSED** [review]

File: HRIS.Services/Services/AuthService.cs [hunk 6]
```diff
@@ -45,13 +49,13 @@ private static bool IsTokenExpired(string token)
...
+        var tokenEndpoint = $"{_authManagement.Issuer ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_ISSUER}oauth/token";
         var requestBody = new Dictionary<string, string>
         {
-            { "client_id",  EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_ID},
-            { "client_secret", EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_SECRET },
+            { "client_id",  _authManagement.ClientId ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_ID},
+            { "client_secret", _authManagement.ClientSecret ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_SECRET },
             { "grant_type", "client_credentials" },
-            { "audience", $"{EnvironmentVariableHelper.AUTH_MANAGEMENT_ISSUER}api/v2/" }
+            { "audience", $"{_authManagement.Issuer ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_ISSUER}api/v2/" }
```
Comment (bseptember) [inline HRIS.Services/Services/AuthService.cs:58]: Refactor: can this entire thing perhaps be a variable set once in the constructor?

string _clientId = _authManagement.ClientId ?? EnvironmentVariableHelper.AUTH_MANAGEMENT_CLIENT_ID;

Same applies to rest of code.

**Review by bseptember — CHANGES_REQUESTED** [review]

**Review by bseptember — DISMISSED** [review]
@EduanO @nzikalala once merged, we need to check **the server** if this breaks. If it does, then all the values must be swapped around:
eg.
this
_smtpSettings.Password ?? EnvironmentVariableHelper.SMTP_PASSWORD
to
EnvironmentVariableHelper.SMTP_PASSWORD ??_smtpSettings.Password
and so on.

**Review by EduanO — APPROVED** [review]

**Review by EduanO — APPROVED** [review]

**Review by MathildaShongwe15 — APPROVED** [review]

**Review by ReitumetseMoloto — APPROVED** [review]

**Pull request merged**